src/main/java/org/alliancegenome/curation_api/jobs/executors/OrthologyExecutor.java

```diff
@@ -0,0 +1,138 @@
+package org.alliancegenome.curation_api.jobs.executors;
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData;
+import org.alliancegenome.curation_api.interfaces.AGRCurationSchemaVersion;
+import org.alliancegenome.curation_api.model.entities.Molecule;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkFMSLoad;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFile;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFileHistory;
+import org.alliancegenome.curation_api.model.entities.orthology.GeneToGeneOrthologyGenerated;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.OrthologyFmsDTO;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.OrthologyIngestFmsDTO;
+import org.alliancegenome.curation_api.response.APIResponse;
+import org.alliancegenome.curation_api.response.LoadHistoryResponce;
+import org.alliancegenome.curation_api.services.orthology.GeneToGeneOrthologyGeneratedService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
+import org.apache.commons.collections.ListUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import io.quarkus.logging.Log;
+import lombok.extern.jbosslog.JBossLog;
+
+@JBossLog
+@ApplicationScoped
+public class OrthologyExecutor extends LoadFileExecutor {
+
+	@Inject
+	GeneToGeneOrthologyGeneratedService generatedOrthologyService;
+
+	public void runLoad(BulkLoadFile bulkLoadFile) {
+		try {
+			BulkFMSLoad fms = (BulkFMSLoad) bulkLoadFile.getBulkLoad();
+			
+			OrthologyIngestFmsDTO orthologyData = mapper.readValue(new GZIPInputStream(new FileInputStream(bulkLoadFile.getLocalFilePath())), OrthologyIngestFmsDTO.class);
+			bulkLoadFile.setRecordCount(orthologyData.getData().size());
+			
+			AGRCurationSchemaVersion version = Molecule.class.getAnnotation(AGRCurationSchemaVersion.class);
```

Comment:
@markquintontulloch shouldn't the version be coming from the Orthology class vs the Molecule class?

Reply:
Yep, good spot - fixed